searx/engines/sqlite.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring
+
+"""SQLite database (Offline)
+
+"""
+
+import sqlite3
+
+from searx import logger
+
+logger = logger.getChild('SQLite engine')
+
+engine_type = 'offline'
+database = ""
+query_str = ""
+limit = 10
+paging = True
+result_template = 'key-value.html'
+
+class SQLiteDB:
+    """
+    Implements a `Context Manager`_ for a SQLite.
+
+    usage::
+
+        with SQLiteDB('test.db') as cur:
+            print(cur.execute('select sqlite_version();').fetchall()[0][0])
+
+    .. _Context Manager: https://docs.python.org/3/library/stdtypes.html#context-manager-types
+    """
+
+    def __init__(self, db):
+        self.database = db
+        self.connect = None
+
+    def __enter__(self):
+        self.connect = sqlite3.connect(self.database)
+        self.connect.row_factory = sqlite3.Row
+        return self.connect.cursor()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type is None:
+            self.connect.commit()
```

Comment:
There is no need to commit for a SELECT : https://docs.python.org/3/library/sqlite3.html#sqlite3-controlling-transactions

Reply:
Not in this use case, but in more general use cases a cursor can execute update statements. I placed the class in the utils section and renamed it:

https://github.com/searxng/searxng/pull/80/files#diff-8d38f1d52f68ec93ec84ebb815fbbb6240fd70d8076dc11ee55021ad58c5658eR617

Reply:
Searx is a (meta) search engine.  Is there a use case for using it to do anything other than query databases?

Reply:
Not at the moment but may be we wish to store some data one day.  This is just a class (context manager) generalized to use fore multi usage, placed in module `utils.py`